core/arch/arm/kernel/boot.c

```diff
@@ -1292,6 +1292,17 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
 	init_sec_mon(nsec_entry);
 }
 
+#if defined(ARM32) && (CFG_TEE_CORE_LOG_LEVEL >= TRACE_DEBUG)
+static bool cpu_nmfi_enabled(void)
+{
+	uint32_t sctlr = 0;
+
+	sctlr = read_sctlr();
+
+	return !!(sctlr & SCTLR_NMFI);
```

Comment:
`return sctlr & SCTLR_NMFI;` should be enough, the C standard already delivers what you try to do with `!!`.

Reply:
OK